invisible_cities/reco/calib_functions_test.py

```diff
@@ -168,4 +174,131 @@ def test_copy_sensor_table3(config_tmpdir):
+    assert_approx_equal(ped_values.gain_min , -538.68814, 5)
+    assert_approx_equal(ped_values.gain_max ,  538.39577, 5)
+    assert_approx_equal(ped_values.sigma_max, 1076.97317, 5)
+    assert_approx_equal(ped_values.sigma_min,      0.001)
+
+
+@mark.parametrize('               file_name, sensor_spe, sensor_dark,    sensor_bins , expected_run_no, expected_n_sensors',
+                (('sipmcalspectra_R6358.h5', 'sipm_spe', 'sipm_dark', 'sipm_spe_bins',            6358,               1792),
+                 ( 'pmtcalspectra_R6354.h5',  'pmt_spe',  'pmt_dark', 'pmt_dark_bins',            6354,                 12)))
+def test_valid_bins_exist( ICDATADIR, file_name, sensor_spe, sensor_dark, sensor_bins, expected_run_no, expected_n_sensors):
+    PATH_IN = os.path.join(ICDATADIR, file_name)
+    h5in    = tb.open_file(PATH_IN, 'r')
+    run_no  = get_run_number(h5in)
+
+    specsL = np.array(getattr(h5in.root.HIST, sensor_spe)).sum(axis=0)
+    specsD = np.array(getattr(h5in.root.HIST, sensor_dark)).sum(axis=0)
+    bins   = np.array(getattr(h5in.root.HIST, sensor_bins))
+
+    min_stat = 10
+
+    for ich, (led, dar) in enumerate(zip(specsL, specsD)):
+        b1 = 0
+        b2 = len(dar)
+        if min_stat != 0:
+            try:
+                valid_bins = np.argwhere(led>=min_stat)
+                b1 = valid_bins[ 0][0]
+                b2 = valid_bins[-1][0]
+            except IndexError:
+                continue
+
+    assert run_no      == expected_run_no
+    assert len(specsL) == expected_n_sensors
+    assert len(specsL) == len(specsD)
+    assert len(bins)   == 349
+
+    assert not (b1 == 0)
+    assert not (b2 == 0)
+
+
+def test_seeds_without_using_db(ICDATADIR):
+    PATH_IN = os.path.join(ICDATADIR, 'sipmcalspectra_R6358.h5')
+    h5in    = tb.open_file(PATH_IN, 'r')
+    run_no  = get_run_number(h5in)
+
+    specsL = np.array(h5in.root.HIST.sipm_spe).sum(axis=0)
+    specsD = np.array(h5in.root.HIST.sipm_dark).sum(axis=0)
+    bins   = np.array(h5in.root.HIST.sipm_spe_bins)
+
+    min_stat = 10
+
+    for ich, (led, dar) in enumerate(zip(specsL, specsD)):
+        b1 = 0
+        b2 = len(dar)
+        if min_stat != 0:
+            try:
+                valid_bins = np.argwhere(led>=min_stat)
+                b1 = valid_bins[ 0][0]
+                b2 = valid_bins[-1][0]
+            except IndexError:
+                continue
+
+        pD = find_peaks_cwt(dar, np.arange(2, 20), min_snr=2)
+        if len(pD) == 0:
+            continue
+
+        gb0     = [(0, -100, 0), (1e99, 100, 10000)]
+        sd0     = (dar.sum(), 0, 2)
+        sel     = np.arange(pD[0]-5, pD[0]+5)
+        errs    = poisson_sigma(dar[sel], default=0.1)
+        gfitRes = fitf.fit(fitf.gauss, bins[sel], dar[sel], sd0, sigma=errs, bounds=gb0)
+
+        ped_vals      = np.array([gfitRes.values[0], gfitRes.values[1], gfitRes.values[2]])
+        scaler_func   = cf.dark_scaler(dar[b1:b2][(bins[b1:b2]>=-5) & (bins[b1:b2]<=5)])
```

Comment:
too many operations in one line, split into many. Maybe define `indices = slice(b1, b2)`. Maybe use `in_range`.